Adds new Design System Tooltip component

martins87 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: #4123 

## Description of Changes
- Adds new Design System Tooltip component
- Adds Storybook stories for the new component

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_tooltip.tsx:
```diff
@@ -0,0 +1,50 @@
+
+  return (
+    <Tooltip
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        justifyContent: 'center',
+        padding: '4px 8px',
+        isolation: 'isolate',
+        maxWidth: '232px',
+        height: '40px',
+        backgroundColor: '#141315',
+        color: '#FFFFFF',
+        borderRadius: '4px',
+        fontFamily: 'NeueHaasUnica',
+        fontStyle: 'normal',
+        fontWeight: 400,
+        fontSize: '12px',
+        lineHeight: '16px',
+        textAlign: 'center',
+        letterSpacing: '0.02em',
+        fontFeatureSettings: `'tnum' on, 'lnum' on`,
+        '& .MuiTooltip-arrow::before': {
+          borderTopColor: '#141315',
+          borderRightColor: '#141315',
+          borderRadius: '0',
+        },
+      }}
```
this is not acceptable. In last review I pointed out that we should not use inline styles

masvelio commented on packages/commonwealth/package.json:
```diff
@@ -113,6 +115,7 @@
     "@metamask/detect-provider": "^2.0.0",
     "@metamask/eth-sig-util": "^4.0.0",
     "@mui/base": "^5.0.0-beta.5",
+    "@mui/joy": "^5.0.0-alpha.87",
```
we dont want another library especially if this is in alpha

masvelio commented on packages/commonwealth/package.json:
```diff
@@ -95,6 +95,8 @@
     "@capacitor/browser": "^4.0.1",
     "@cosmjs/amino": "^0.26.1",
     "@cosmjs/proto-signing": "^0.26.1",
+    "@emotion/react": "^11.11.1",
+    "@emotion/styled": "^11.11.0",
```
big no :<

masvelio reviewed (changes requested):
- According to AC, it should display max of 2 lines. Probably using `...` by the end of second. 
- if one line it looks bad
https://github.com/hicommonwealth/commonwealth/assets/14819225/adf90f67-da84-450a-985a-8d15c285a008
- we dont want to use mui joi

martins87 commented:
Many thanks for the PR review, @masvelio!

I'll address them all and go with the Popper implementation.
Adding another library just because mui-base doesn't have the component yet is not a good choice.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_tooltip.tsx:
```diff
@@ -0,0 +1,75 @@
+  );
+};
+
+export const CWTooltip = (props: TooltipProps) => {
```
destructure props right away and please keep that in mind in the next PRs.

```suggestion
export const CWTooltip = ({ content, renderTrigger, placement }: TooltipProps) => {
```

martins87 replied:
Addressed. Noted.

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_tooltip.tsx:
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+import 'components/component_kit/new_designs/cw_tooltip.scss';
```
please change the name of file to `CWTooltip.tsx` and `CWTooltip.scss` and keep that apprach in the upcoming PRs

martins87 replied:
Addressed, noted.

masvelio commented on packages/commonwealth/client/styles/components/component_kit/new_designs/cw_tooltip.scss:
```diff
@@ -0,0 +1,84 @@
+  padding: 4px 8px;
+  isolation: isolate;
+  max-width: 232px;
+  height: '40px';
```
40px not '40px'

martins87 replied:
Addressed

masvelio commented on packages/commonwealth/client/styles/components/component_kit/new_designs/cw_tooltip.scss:
```diff
@@ -0,0 +1,84 @@
+  }
+
+  .Text {
+    text-align: center;
+    font-family: 'NeueHaasUnica';
+    font-style: 'normal';
+    font-weight: 400;
+    font-size: '12px';
+    line-height: '16px';
+    text-align: 'center';
+    letter-spacing: '0.02em';
+    font-feature-settings: 'tnum' on, 'lnum' on;
+    color: $white;
```
please do not copy styles from figma without validating it. 
- double `text-align`
- do not use `'` where this does not supposed to be 

![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/8be95fd3-14c2-4bb1-8792-9ba4cbc8cfb0)

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/cw_component_showcase.tsx:
```diff
@@ -273,7 +274,7 @@ export const ComponentShowcase = () => {
         <div className="tooltip-row">
           <CWText>Solid background</CWText>
-          <CWTooltip
+          <CWTooltipOld
```
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/435ec053-ce8b-4714-8d42-9b92a116b5d8)

Despite the fact you changed the component to old, the old tooltips looks like the new one. This is because you are breaking the encapsulation rule for styles, which I explain in the other comment

martins87 replied:
Addressed

masvelio commented on packages/commonwealth/client/styles/components/component_kit/new_designs/cw_tooltip.scss:
```diff
@@ -0,0 +1,84 @@
+  left: $left;
+}
+
+.Tooltip {
```
We already have class `Tooltip` in the code so when you have two Tooltips in the page (the old one and the new one, the styles are clashed). My suggestion is that you should change `Tooltip` class into `OldTooltip` in  old `cw_popover/cw_tooltip.scss`

martins87 replied:
Nice catch. Thanks for this!

martins87 replied:
Addressed

masvelio commented on packages/commonwealth/client/styles/components/component_kit/new_designs/cw_tooltip.scss:
```diff
@@ -0,0 +1,84 @@
+  }
+}
+
+.top {
+  margin-bottom: 8px;
+}
+
+.right {
+  margin-left: 8px;
+}
+
+.bottom {
+  margin-top: 8px;
+}
+
+.left {
+  margin-right: 8px;
+}
```
same problem here, doing this you make the encapsulation useless so these classes can mess up other components unexpectedly. Move it inside main `.Tooltip` class.

martins87 replied:
Addressed

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_tooltip.tsx:
```diff
@@ -0,0 +1,75 @@
+      {renderTrigger(popoverProps.handleInteraction)}
+      <Popover
+        placement={placement}
+        content={
+          typeof content === 'string' ? (
+            <Container placement={placement}>
+              <CWText type="caption">{formatText(content)}</CWText>
+            </Container>
+          ) : (
+            <Container placement={placement}>{content}</Container>
+          )
+        }
```
It can be simplified
```suggestion
          content={
            <Container placement={placement}>
              <CWText type="caption">
                {content === 'string' ? formatText(content) : content}
              </CWText>
            </Container>
          }
```

martins87 replied:
Addressed

martins87 commented:
> * According to AC, it should display max of 2 lines. Probably using `...` by the end of second.
> * if one line it looks bad
> 
>  Kapture.2023-07-18.at.12.19.47.mp4 
> * we dont want to use mui joi

@zakhap what is the expected behaviour when the text is larger than like 70 characters / two lines?

Currently I added ... as Marcin suggested.

zakhap commented:
> > * According to AC, it should display max of 2 lines. Probably using `...` by the end of second.
> > * if one line it looks bad
> > 
> > Kapture.2023-07-18.at.12.19.47.mp4
> > 
> > * we dont want to use mui joi
> 
> @zakhap what is the expected behaviour when the text is larger than like 70 characters / two lines?
> 
> Currently I added `...` as Marcin suggested.

I agree with Marcin that the component should handle excess with an ellipsis. Make sure this is using css overflow ellipsis rather than some sort of text truncation function.

masvelio commented:
> > > * According to AC, it should display max of 2 lines. Probably using `...` by the end of second.
> > > * if one line it looks bad
> > > 
> > > Kapture.2023-07-18.at.12.19.47.mp4
> > > 
> > > * we dont want to use mui joi
> > 
> > 
> > @zakhap what is the expected behaviour when the text is larger than like 70 characters / two lines?
> > Currently I added `...` as Marcin suggested.
> 
> I agree with Marcin that the component should handle excess with an ellipsis. Make sure this is using css overflow ellipsis rather than some sort of text truncation function.

we have a mixin for than `multiline-text-ellipsis`

martins87 commented:
> > > > * According to AC, it should display max of 2 lines. Probably using `...` by the end of second.
> > > > * if one line it looks bad
> > > > 
> > > > Kapture.2023-07-18.at.12.19.47.mp4
> > > > 
> > > > * we dont want to use mui joi
> > > 
> > > 
> > > @zakhap what is the expected behaviour when the text is larger than like 70 characters / two lines?
> > > Currently I added `...` as Marcin suggested.
> > 
> > 
> > I agree with Marcin that the component should handle excess with an ellipsis. Make sure this is using css overflow ellipsis rather than some sort of text truncation function.
> 
> we have a mixin for than `multiline-text-ellipsis`

Addressed.

masvelio commented:
@martins87 thanks for addressing my comments. 

The tooltips I have introduced [in this PR](https://github.com/hicommonwealth/commonwealth/pull/4513) are broken now (they are white instead of black).
 
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/0deab171-ee1e-46a4-b8d0-4edb23e6131f)
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/9c9e066d-ca9e-4bca-bfce-dff263a517e2)

You can fix these my changing `.Tooltip` to `.OldTooltip` [here](https://github.com/hicommonwealth/commonwealth/blob/708de98bb03de1b7bd7fd811662adeee17622b63/packages/commonwealth/client/scripts/views/pages/discussions/CommentCard/CommentCard.scss#L98) and [here](https://github.com/hicommonwealth/commonwealth/blob/708de98bb03de1b7bd7fd811662adeee17622b63/packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/ThreadOptions.scss#L8)

Please fix that. Otherwise looks good I am approving this ✅

masvelio commented:
One more thing. Please make sure that your PRs have proper `Description of Changes`

martins87 commented:
> @martins87 thanks for addressing my comments.
> 
> The tooltips I have introduced [in this PR](https://github.com/hicommonwealth/commonwealth/pull/4513) are broken now (they are white instead of black).
> 
> ![image](https://user-images.githubusercontent.com/14819225/256206752-0deab171-ee1e-46a4-b8d0-4edb23e6131f.png) ![image](https://user-images.githubusercontent.com/14819225/256206826-9c9e066d-ca9e-4bca-bfce-dff263a517e2.png)
> 
> You can fix these my changing `.Tooltip` to `.OldTooltip` [here](https://github.com/hicommonwealth/commonwealth/blob/708de98bb03de1b7bd7fd811662adeee17622b63/packages/commonwealth/client/scripts/views/pages/discussions/CommentCard/CommentCard.scss#L98) and [here](https://github.com/hicommonwealth/commonwealth/blob/708de98bb03de1b7bd7fd811662adeee17622b63/packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/ThreadOptions.scss#L8)
> 
> Please fix that. Otherwise looks good I am approving this ✅

Addressed.

martins87 commented:
> 

Noted. Thanks again for the review on this one.

masvelio commented:
@jnaviask let's merge this in
